#**********************************************************************
# Copyright 2020 Advanced Micro Devices, Inc
# Licensed under the Apache License, Version 2.0 (the "License");
# you may not use this file except in compliance with the License.
# You may obtain a copy of the License at
# 
#     http://www.apache.org/licenses/LICENSE-2.0
# 
# Unless required by applicable law or agreed to in writing, software
# distributed under the License is distributed on an "AS IS" BASIS,
# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
# See the License for the specific language governing permissions and
# limitations under the License.
#********************************************************************

# oteodoro -- Added changes disabled install of wheel and boto \
#   and allow user to install them though the package manager instead.

----
--- a/src/rprblender/utils/install_libs.py.orig	2021-10-11 08:25:46.000000000 -0700
+++ b/src/rprblender/utils/install_libs.py	2021-11-09 08:01:09.518824180 -0800
@@ -12,6 +12,13 @@
 # See the License for the specific language governing permissions and
 # limitations under the License.
 #********************************************************************
+
+# oteodoro changed this to disable downloading wheel and boto3 and it's \
+#   dependencies.  It is preferred to do it though the package manager \
+#   (with emerge) instead.
+
+import platform                                                               # oteodoro added this line
+
 import sys
 import site
 import subprocess
@@ -21,6 +28,9 @@ import bpy
 from . import IS_MAC, IS_LINUX, package_root_dir
 from rprblender import config
 
+if 'Linux' == platform.system():                                              # oteodoro added this line
+    import distro                                                             # oteodoro added this line
+
 from rprblender.utils.logging import Log
 log = Log(tag="install_libs")
 
@@ -53,6 +63,11 @@ def ensure_boto3():
     Use this snippet to install boto3 library with all the dependencies if absent at the addon launch time
     Note: still it will be available at the next Blender launch only
     """
+
+    if IS_LINUX:                                                               # oteodoro added this line
+        if distro.id() == "gentoo":                                            # oteodoro added this line
+            return                                                             # oteodoro added this line
+
     pip_check_file = package_root_dir() / PIP_CHECK_FILENAME
 
     try:
